src/plugin_system.js

```diff
@@ -2186,7 +2186,128 @@ const PluginSystem = {
+            if (cur === "0") {
+              return acc
+            } else if (cur === "1" && 軸数字[ src.length -1 - idx ] !== "") {
+              return acc + 軸数字[ src.length -1 - idx ]
+            } else {
+              return acc + replacer(cur) + 軸数字[ src.length -1 - idx ]
+            }
+          }, "")
+          acc = (unit ? unit + 単位数字[idx] : "")  + acc
+          return acc
+        }, "")
+      }
+      return converter(separater(input))
+    }
+  },
+  '算用数字':{
+    type: 'func',
+    josi: [['を','の']],
+    fn: function (input) {
+      function converter (src) {
+        const multibytes = 単位数字.filter( a => a.length > 1 )
+        const result = []
+        for (let idx = 0; idx < src.length; idx++) {
+          const cur = src.slice(idx, idx + 1)
+          const 指数 = 単位数字.includes(cur) ? cur : multibytes.find(v => v === src.slice(idx, idx + v.length))
+          const 位 = 軸数字.includes(cur) ? cur : null
+          const 底 = 基本漢数字.includes(cur) ? cur : null
+          if (指数) {
+            if (指数.length > 1) {
+              idx += 指数.length - 1
+            }
+            result.push(BigInt("1" + "0".repeat(4 * 単位数字.indexOf(指数))))
+          } else if (位) {
+            result.push(10 ** (軸数字.indexOf(位)))
+          } else if (底) {
+            result.push(基本漢数字.indexOf(底))
+          } else {
+            throw new Error('『算用数字』命令の中に無効な文字が含まれています。')
+          }
+        }
+        return result
+      }
+      function separater (arr) {
+        let base = []
+        let unit = []
+        return arr.reduce(( acc, cur, idx ) => {
+          if (cur > 1000) {
+            if(base.length === 0) base.push(0,1)
+            if(base.length === 1) base.push(1)
+            unit.push(base)
+            base = []
+            unit.push(cur)
+            acc.push(unit)
+            unit = []
+          } else if (10 <= cur && cur <= 1000) {
+            if(base.length === 0) base.push(1)
+            base.push(cur)
+            unit.push(base)
+            base = []
+          } else if (cur < 10) {
+            base.push(cur)
+          }
+          if (idx + 1 === arr.length) {
+            if(base.length === 1) {
+              base.push(1)
+              unit.push(base)
+            }
+            acc.push(unit)
+          }
+          return acc
+        }, [])
+      }
+      function calculator (arr) {
+        return arr.reduce(( acc, cur, idx ) => {
+          return acc + cur.reduce((acc, cur, idx) => {
+            return cur > 1000 ? acc * cur : acc + BigInt(cur[0] * cur[1])
+          }, 0n)
+        }, 0n)
+      }
+      const tmp = calculator(separater(converter(input.toString())))
+      return tmp > Number.MAX_SAFE_INTEGER ? tmp : Number(tmp)
+    }
   }
 }
+const 単位数字 = [
+  "", "万", "億", "兆", "京", "垓", "𥝱", "穣", "溝", "澗", "正",
+  "載", "極", "恒河沙", "阿僧祇", "那由他", "不可思議", "無量大数"
+]
+const 軸数字 = [
+  "", "十", "百", "千"
+]
+const 基本算用数字 = "0123456789".split("")
```

Comment:
```suggestion
const 基本算用数字 = [...Array(10).keys()].map(i=>i.toString())
```

こういった形で記述すると漏れが発生しにくくなって良さそうです。